src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/Uninstall/DeleteDirectoriesTask.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.IO.Abstractions;
+using System.Linq;
+using Elastic.Installer.Domain.Configuration.Wix.Session;
+using Elastic.Installer.Domain.Model.Elasticsearch;
+using Elastic.Installer.Domain.Model.Elasticsearch.Locations;
+
+namespace Elastic.InstallerHosts.Elasticsearch.Tasks.Uninstall
+{
+	/// <summary>
+	/// Ensures we remove the Elasticsearch installation directory completely.
+	/// We can't rely on the MSI RemoveFiles step to remove directories/files that were registered 
+	/// on a full uninstall because new files may have been introduced later on (e.g. plugins).
+	/// </summary>
+	public class DeleteDirectoriesTask : ElasticsearchInstallationTaskBase
```

Comment:
`UninstallDeleteDirectoriesTask`

Reply:
I think `DeleteDirectoriesTask` is descriptive enough?

Reply:
renamed to `UninstallDirectoriesTask` to match the custom action